ex5B/ex5.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Write a function that given a string, returns the first character that
+ * doesn't repeat in the whole string. Otherwise (all characters repeat),
+ * should return null.
+ *
+ * For example:
+ * firstUniqueChar("abcdedcba") === "e"
+ * firstUniqueChar("abcdeedcba") === null
+ * firstUniqueChar("abababcab") === "c"
+ * firstUniqueChar("aaaaaaaaa") === null
+ * firstUniqueChar("a") === "a"
+ */
+function firstUniqueChar(str) {
+
+	var objCharCounte = buildCharIndicesMap(str);
+
+	return findFirstUniqueCharInMap(objCharCounte);
+}
+
+function buildCharIndicesMap(str) {
+
+	var objCharCounter = {},  searchIndex = 0, currentChar = '';
+
+	for (searchIndex = 0; searchIndex < str.length; searchIndex++) {
+
+		currentChar = str.charAt(searchIndex);
+
+		if (objCharCounter[currentChar] === undefined) {
+
+			// Create new property
+			objCharCounter[currentChar] = { firstIndex: searchIndex, count: 1};
+
+		} else {
+
+			// Add to property
+			objCharCounter[currentChar].count += 1;
+
+		}
+
+	}
+	return objCharCounter;
+}
+
+function findFirstUniqueCharInMap(objCharCounter) {
+	
+	var charArray = [], objFirstChar = { charVal: null }, currentChar, currentCounter = 0, currentFirstIndex = 0;
```

Comment:
As I said before, you can probably simplify the initial value you have for `objFirstChar` and have it be `var objFirstChar = null`. You'll have to change the if condition below too, obviously.